plantdescriptionengine/src/main/java/eu/arrowhead/core/plantdescriptionengine/providedservices/pde_monitor/routehandlers/GetAllPdeAlarms.java

```diff
@@ -0,0 +1,193 @@
+            .min(0)
+            .build();
+        final IntParameter pageParam = new IntParameter.Builder()
+            .name(PAGE)
+            .min(0)
+            .requires(itemPerPageParam)
+            .build();
+
+        final StringParameter sortFieldParam = new StringParameter.Builder()
+            .name(SORT_FIELD)
+            .legalValues(ID, RAISED_AT, UPDATED_AT, CLEARED_AT)
+            .build();
+        final StringParameter directionParam = new StringParameter.Builder()
+            .name(DIRECTION)
+            .legalValues(ASC, DESC)
+            .defaultValue(ASC)
+            .build();
+        final StringParameter systemNameParam = new StringParameter.Builder()
+            .name(SYSTEM_NAME)
+            .build();
+        final StringParameter severityParam = new StringParameter.Builder()
+            .name(SEVERITY)
+            .legalValues(severityValues)
+            .build();
+        final BooleanParameter acknowledgedParam = new BooleanParameter.Builder()
+            .name(ACKNOWLEDGED)
+            .build();
+
+        final List<QueryParameter> acceptedParameters = List.of(pageParam, sortFieldParam, directionParam,
+            systemNameParam, systemNameParam, severityParam, acknowledgedParam);
+
+        final QueryParamParser parser;
+
+        try {
+            parser = new QueryParamParser(null, acceptedParameters, request);
+        } catch (final ParseError error) {
+            logger.error("Encountered the following error(s) while parsing an HTTP request: " + error.getMessage());
+            response
+                .status(HttpStatus.BAD_REQUEST)
+                .body(ErrorMessage.of(error.getMessage()), CodecType.JSON);
+            return Future.success(response);
+        }
+
+        List<PdeAlarmDto> alarms = alarmManager.getAlarms();
+
+        final Optional<String> sortField = parser.getValue(sortFieldParam);
+        if (sortField.isPresent()) {
+            final String sortDirection = parser.getRequiredValue(directionParam);
+            final boolean ascending = ASC.equals(sortDirection);
+
+            switch (sortField.get()) {
+                case ID:
+                    PdeAlarm.sortById(alarms, ascending);
+                    break;
+                case RAISED_AT:
+                    PdeAlarm.sortByRaisedAt(alarms, ascending);
+                    break;
+                case UPDATED_AT:
+                    PdeAlarm.sortByUpdatedAt(alarms, ascending);
+                    break;
+                case CLEARED_AT:
+                    PdeAlarm.sortByClearedAt(alarms, ascending);
+                    break;
+                default:
+                    // We should never reach this case, since the sortField
+                    // param has been validated by the parser.
+                    throw new AssertionError("Encountered the invalid sort field '" + sortField + "'.");
+
+            }
+        }
+
+        final Optional<Integer> page = parser.getValue(pageParam);
```

Comment:
filtering should always precedes paging.

Reply:
Fixed